components/num-util/src/fixed_decimal.rs

```diff
@@ -519,3 +699,31 @@ fn test_lower_magnitude_bounds() {
     assert_eq!(Error::Limit, dec.multiply_pow10(-1).unwrap_err());
```

Comment:
In `test_upper_magnitude_bounds` and `test_lower_magnitude_bounds`, check FromStr round-tripping on every valid `dec` that gets created.  It will be an easy way to verify the boundary conditions.